WordPress/Classes/Models/Post.swift

```diff
@@ -195,7 +195,7 @@ class Post: AbstractPost {
 
         if let originalPost = original as? Post {
 
-            if tags?.characters.count != originalPost.tags?.characters.count && tags != originalPost.tags {
+            if tags ?? "" != originalPost.tags ?? "" {
```

Comment:
You've almost convinced me ;) On top of what you said separating it in methods will also help to improve testing. One consideration about splitting it in methods is that all of them would have to have the following guard right?:

`if let originalPost = original as? Post {`

Swift definitely has a way of making easy things complicated.